mmeval/metrics/anet_ar.py

```diff
@@ -0,0 +1,318 @@
+        >>>                {'video_name': 'v_--1DO2V4K74',
+        >>>                 'annotations': [
+        >>>                     {'segment': [30.02, 180], 'label': 1}],
+        >>>                 'proposal_list': [
+        >>>                     {'segment': [53, 68], 'score': 0.21},
+        >>>                     {'segment': [38, 110], 'score': 0.54},
+        >>>                     {'segment': [65, 128], 'score': 0.95},
+        >>>                     {'segment': [69, 93], 'score': 0.98},
+        >>>                     {'segment': [99, 147], 'score': 0.84},
+        >>>                     {'segment': [28, 96], 'score': 0.84},
+        >>>                     {'segment': [18, 92], 'score': 0.22},
+        >>>                     {'segment': [40, 66], 'score': 0.36},
+        >>>                     {'segment': [14, 29], 'score': 0.75},
+        >>>                     {'segment': [67, 105], 'score': 0.25},
+        >>>                     {'segment': [2, 7], 'score': 0.94},
+        >>>                     {'segment': [25, 112], 'score': 0.49},
+        >>>                     {'segment': [7, 83], 'score': 0.9},
+        >>>                     {'segment': [75, 159], 'score': 0.42},
+        >>>                     {'segment': [99, 176], 'score': 0.62},
+        >>>                     {'segment': [89, 186], 'score': 0.56},
+        >>>                     {'segment': [50, 200], 'score': 0.5}]
+        >>>                 }]
+        >>> anet_metric(predictions)
+        OrderedDict([('auc', 54.2), ('AR@1', 0.0), ('AR@5', 0.0),
+                     ('AR@10', 0.2), ('AR@100', 0.6)])
+    """
+
+    def __init__(self,
+                 temporal_iou_thresholds: Union[np.array, List] = np.linspace(
+                     0.5, 0.95, 10),
+                 max_avg_proposals: int = 100,
+                 **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.temporal_iou_thresholds = np.array(temporal_iou_thresholds)
+        self.max_avg_proposals = max_avg_proposals
+        self.ground_truth: Dict[str, np.array] = {}
+
+    def add(self, predictions: Sequence[dict]) -> None:  # type: ignore
+        """Add detection results to the results list.
+
+        Args:
+            predictions (Sequence[dict]): A list of prediction dict which
+                contains the following keys:
+
+                - `video_name`: The name of the video, e.g., `v_--1DO2V4K74`.
+                   The name should be of the format of `v_{video_id}`.
+                - `annotations`: A list of annotations. Each annotation is a
+                  dict with keys `segment` and `label`. The value of `segment`
+                  is `[start, end]`. The value of `label` is an int label.
+                - `proposal_list`: A list of proposals. Each proposal is a
+                  dict with keys `segment` and `score`. The value of `segment`
+                  is `[start, end]`. The value of `score` is a float number.
```

Comment:
I'm slightly confused about this function. So `groudtruths` are actually in `predictions`? Is there any reason why not using `add(predictions, groundtruths)`?

Reply:
Yes groudtruths are in predictions. Not specific reason, just followed the implementation from mmaction2.

Reply:
@hukkai In MMEval we are providing metrics to more users, so separating `groundtruths` from `predictions` should be more reasonable. Should do some adaption in mmaction2 later. ^_^

Reply:
@C1rN09 Hi, I have refactored this part. Could you have a look? Thanks!